### ISSUE TITLE
[Wpf] No Info event fired with a stylus (and touch)

### ISSUE THREAD

**Issue by Aruchu** [pauldendulk/Mapsui#200]
Hi,

I am currently using the WPF library to develop a map with clickable features on it, using the Info event. However, our software will be deployed on Windows tablets equipped with stylus.
As it is, the WPF lib ignores events if they come from a stylus. This feature seems to be hard-coded on purpose in **Mapsui.UI.Wpf/MapControl.cs**, since `MapControlMouseLeftButtonDown(...)` and `MapControlMouseLeftButtonUp(...)` start with `if (e.StylusDevice != null) return;`.

We traced these lines back to commit b37b211ec3767da71edf68139a2372c50d2689f9, but couldn't understand the reason behind it.
So I tried simply commenting this line in both handlers, while preserving it in `MapControlMouseMove(...)`, and I couldn't notice any side effect.

As the Info event is unusable on tablet right now, and unless we are overlooking a major issue, would you accept a pull request removing this limitation?

PS: since `if (e.StylusDevice != null) return;` is in `MapControlMouseMove(...)` as well, some "CanMoveMapWithStylus" parameter could also be added to MapControl for the developer to decide whether he needs this.

**Comment by Aruchu** [comment]
Another precision, we would also like to fire the Info event after a touch on a feature. This, however, would require, as I understand it, subscribing to TouchDown and TouchUp and giving the handlers a similar behavior as their mouse handler counterparts.

**Comment by pauldendulk** [comment]
The check on StylusDevice was added because it caused double touch events. This was a long time ago. The implementation on StylusDevice might have changed. I have to take some time to look into this. In which way could I test StylusDevice without having the hardware?

**Comment by pauldendulk** [comment]
Your second comment suggests different handlers for mouse and stylus. Could this not be dealt with with a single mechanism? 

Also, if you simple remove the stylus check in code, does this solve your problem?

**Comment by Skymirrh** [comment]
Hi @pauldendulk, thanks for answering.

>Your second comment suggests different handlers for mouse and stylus. Could this not be dealt with with a single mechanism?

AFAIK different handlers aren't needed unless specific interactions are required, because if Stylus/Touch events are not handled via specific Stylus/Touch handlers, they are passed down to the Mouse handler, as follow (console output of a small WPF app made to check input event firing order, here for a Touch input):
```
PreviewStylusDown
StylusDown
PreviewStylusButtonDown
StylusButtonDown
PreviewTouchDown
TouchDown
PreviewMouseLeftButtonDown
PreviewMouseDown
MouseLeftButtonDown
MouseDown
PreviewStylusUp
StylusUp
PreviewStylusButtonUp
StylusButtonUp
PreviewTouchUp
TouchUp
PreviewMouseLeftButtonUp
PreviewMouseUp
MouseLeftButtonUp
MouseUp
```

>Also, if you simple remove the stylus check in code, does this solve your problem?

In fact, that's what we did. We removed the checks and created an internal NuGet package in order to deploy an application on time, and overall it seems to work well. We have other issues with Touch events not being properly fired for specific items on the map, but I don't think this is related to MapsUI -- we will investigate this at a later time, so in case we find anything we will notify you.

>In which way could I test StylusDevice without having the hardware?

FYI, there is an touch input simulator bundled with Visual Studio. More information on how it works in this [Visual Studio blog post](https://blogs.msdn.microsoft.com/visualstudio/2011/09/29/first-look-at-windows-simulator/).

The simulator is located here: `C:\Program Files (x86)\Common Files\Microsoft Shared\Windows Simulator\<visual_studio_version>\Microsoft.Windows.Simulator.exe`. It lets you simulate Touch input and Manipulation input (pinch/zoom and rotation). I usually fire it up, launch a process in there, and then attach that process to Visual Studio in order to debug Touch input.

As we can see from the trace above, Touch input actually fires Stylus events as well.
